cmd/process-agent/subcommands/check/check.go

```diff
@@ -73,26 +74,35 @@ func nextGroupID() func() int32 {
 
 // Commands returns a slice of subcommands for the `check` command in the Process Agent
 func Commands(globalParams *command.GlobalParams) []*cobra.Command {
+	desc := "Run a specific check and print the results. Choose from: process, rtprocess, container, rtcontainer, connections, process_discovery, process_events"
+	return []*cobra.Command{MakeCommand(globalParams, "check", desc)}
+}
+
+func MakeCommand(globalParams *command.GlobalParams, name string, short string) *cobra.Command {
 	cliParams := &cliParams{
 		GlobalParams: globalParams,
 	}
 
 	checkCmd := &cobra.Command{
-		Use:   "check",
-		Short: "Run a specific check and print the results. Choose from: process, rtprocess, container, rtcontainer, connections, process_discovery, process_events",
+		Use:   name,
+		Short: short,
 
 		Args: cobra.ExactArgs(1),
 		RunE: func(cmd *cobra.Command, args []string) error {
 			cliParams.checkName = args[0]
 
+			if !strings.Contains(short, cliParams.checkName) {
```

Comment:
Hmmm, this condition will also pass if the checkName is "specific" or "print" or "results". Should it be stricter?

```
var allowList = []string{"process", "rtprocess", "container", ...}
...
if !slices.Contains(allowList, cliParams.checkName) {
```

Reply:
This would be caught as there is another check during the commands execution. However, I still added the allowlist as its more straightforward.